corehq/apps/repeaters/repeater_generators.py

```diff
@@ -55,8 +55,8 @@ class FormRepeaterXMLPayloadGenerator(BasePayloadGenerator):
     def get_payload(self, repeat_record, payload_doc):
         return payload_doc.get_xml()
 
-    def get_test_payload(self):
-        return self.get_payload(None, _get_test_form())
+    def get_test_payload(self, domain=None):
+        return self.get_payload(None, _get_test_form(domain))
```

Comment:
Note that here `self.repeater.domain` is not set, so the domain has to be passed all the from view